arangod/Aql/Ast.cpp

```diff
@@ -2073,6 +2073,38 @@ void Ast::validateAndOptimize() {
   _root = traverseAndModify(_root, preVisitor, visitor, postVisitor);
 }
 
+/// @brief determines the variables referenced in an expression
```

Comment:
I don't see any need for this method.
We have the same method that takes an arangodb::HashSet as its second parameter. Can't we just use that?

Reply:
without this there is a build failure http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/3959/

Reply:
you have to merge the recent devel into your branch then